pkg/network/setup/netpod/discover.go

```diff
@@ -134,6 +142,123 @@ func (n NetPod) storePodInterfaceData(vmiSpecIface v1.Interface, ifaceState nmst
 	return nil
 }
 
+func (n NetPod) storeBridgeBindingDHCPInterfaceData(currentStatus *nmstate.Status, podIfaceStatus nmstate.Interface, vmiSpecIface v1.Interface, podIfaceName string) error {
+	var dhcpConfig cache.DHCPConfig
+	dhcpConfig.IPAMDisabled = true
+	if ipAddress := firstIPGlobalUnicast(podIfaceStatus.IPv4); ipAddress != nil {
+		dhcpConfig.IPAMDisabled = false
+
+		addr, iperr := vishnetlink.ParseAddr(fmt.Sprintf("%s/%d", ipAddress.IP, ipAddress.PrefixLen))
+		if iperr != nil {
+			return iperr
+		}
+		dhcpConfig.IP = *addr
+
+		mac, err := resolveMacAddress(podIfaceStatus.MacAddress, vmiSpecIface.MacAddress)
+		if err != nil {
+			return err
+		}
+		dhcpConfig.MAC = mac
+
+		linkRoutes, err := filterIPv4RoutesByInterface(currentStatus, podIfaceName)
+		if err != nil {
+			return err
+		}
+		dhcpConfig.Gateway = net.ParseIP(linkRoutes[0].NextHopAddress)
+
+		otherRoutes, err := filterRoutesByNonLocalDestination(linkRoutes, addr)
```

Comment:
I have to admit it is hard for me to understand this routing filtering logic. It is different than the original logic, can you please explain why?